src/library_gl.js

```diff
@@ -68,16 +68,23 @@ var LibraryGL = {
        } */
 
     stringCache: {},
-#if USE_WEBGL2
+#if USE_WEBGL2 || USE_WEBGL2_COMPUTE
     stringiCache: {},
 #endif
     tempFixedLengthArray: [],
 
     packAlignment: 4,   // default alignment is 4 bytes
     unpackAlignment: 4, // default alignment is 4 bytes
 
+    // WebGL 2.0 Compute is deemed to be a design mistake in naming. The
+    // following constants will make the life easier.
+    WEBGL2_COMPUTE_MAJOR_VERSION: 2,
+    WEBGL2_COMPUTE_MINOR_VERSION: 1,
```

Comment:
These two lines would introduce regression on pages that target WebGL 2.0 only, but not WebGL 2.0 Compute. Btw about the comment here - is there a future direction to rename WebGL 2.0 Compute to WebGL 2.1 officially? Or how does the comment relate to current thinking? (If the API name "WebGL 2.0 Compute" is deemed to be permanent, we do not want to retain judgement in comments, so could leave that out, even if we do feel it was the wrong call)

Reply:
As Apple stops to advance OpenGL, WebGL 2.0 Compute can't run on macOS/iOS. Given it's not a general solution for all platforms, the decision from Khronos WebGL Working Group was to make it an extension, rather than a concrete version.

Reply:
> And can you elaborate more on the regression? Is it about code size?

Oh sorry, yeah, meant here a regression on code size specifically.

> I do not think WebGL 2.0 Compute will be renamed to WebGL 2.1 or 3.0 in the near future.

If WebGL 2.0 Compute will not be renamed to WebGL 2.1, but we still have items like

```
    WEBGL2_COMPUTE_MAJOR_VERSION: 2,
    WEBGL2_COMPUTE_MINOR_VERSION: 1,
```

making Emscripten treat WebGL 2.0 Compute as if it was WebGL 2.1, why should we diverge from them and call it 2.1 if Khronos WebGL WG does not? Is there a danger of the working group introducing a "real" WebGL 2.1 at some point after this, and then we would be left with a collision with version numbering?

Reply:
The WebGL version number is heavily used in `library_gl.js` for comparison, so I suppose WebGL 2.0 Compute also needs a numerical representation for this, but it looks inappropriate to just use `2.0`. A real WebGL 2.1 could be the risk, and I haven't got a better idea yet.